spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/RecordMessagingMessageListenerAdapter.java

```diff
@@ -71,8 +72,17 @@ public RecordMessagingMessageListenerAdapter(Object bean, Method method, KafkaLi
 	 */
 	@Override
 	public void onMessage(ConsumerRecord<K, V> record, Acknowledgment acknowledgment, Consumer<?, ?> consumer) {
-		Message<?> message = toMessagingMessage(record, acknowledgment, consumer);
-		logger.debug(() -> "Processing [" + message + "]");
+		Message<?> message;
+		if (isConversionNeeded()) {
+			message = toMessagingMessage(record, acknowledgment, consumer);
+		}
+		else {
+			message = NULL_MESSAGE;
+		}
+		if (logger.isDebugEnabled()) {
+			final Message<?> toLog = message;
+			logger.debug(() -> "Processing [" + toLog + "]");
```

Comment:
Or we don't need `isDebugEnabled()` or we don't need `toLog` variable and shouldn't use `Supplier`-based `debug()`